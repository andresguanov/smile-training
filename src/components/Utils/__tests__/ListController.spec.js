import { shallowMount, mount } from "@vue/test-utils"
import smListController from "@components/Utils/ListController"

let wrapper

describe("smListController", () => {
  beforeEach(() => {
    wrapper = mount(smListController, {
      propsData: {
        listLength: 3
      },
      scopedSlots: {
        default(props) {
          const content = [0, 1, 2].map(i =>
            this.$createElement("span", {
              attrs: { id: "s" + i, selected: props.selectedIndex === i }
            })
          )
          return this.$createElement("div", content)
        }
      }
    })
  })
  //#region Basic render
  it("renders default smListController", () => {
    expect(wrapper.findAll("span").wrappers.length).toBe(3)
    expect(wrapper.html()).toMatchSnapshot()
  })
  //#endregion

  //#region Props
  //#endregion

  //#region Methods
  it("selects and clears index", async () => {
    await wrapper.vm.setSelectedIndex(0)
    expect(wrapper.find("#s0").attributes("selected")).toBe("selected")
    expect(wrapper.html()).toMatchSnapshot()
    await wrapper.vm.clearSelectedIndex()
    expect(wrapper.find("#s0").attributes("selected")).toBeFalsy()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("handles enter", async () => {
    const selected = jest.fn()
    const preventDefault = jest.fn()
    wrapper = mount(smListController, {
      propsData: {
        listLength: 3
      },
      scopedSlots: {
        default(props) {
          const content = [0, 1, 2].map(i =>
            this.$createElement("span", {
              attrs: { id: "s" + i, selected: props.selectedIndex === i }
            })
          )
          return this.$createElement("div", content)
        }
      },
      listeners: {
        selected
      }
    })
    await wrapper.vm.setSelectedIndex(0)
    await wrapper.vm.handleEnter({ preventDefault })
    expect(selected).toHaveBeenCalledWith(0)
    expect(selected).toHaveBeenCalledTimes(1)
    expect(preventDefault).toHaveBeenCalledTimes(1)
  })
  //#endregion

  //#region Events
  //#endregion

  //#region Slots
  //#endregion
})
