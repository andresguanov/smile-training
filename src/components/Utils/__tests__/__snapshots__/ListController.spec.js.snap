// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smListController renders default smListController 1`] = `
<div>
  <div><span id="s0"></span><span id="s1"></span><span id="s2"></span></div>
</div>
`;

exports[`smListController selects and clears index 1`] = `
<div>
  <div><span id="s0" selected="selected"></span><span id="s1"></span><span id="s2"></span></div>
</div>
`;

exports[`smListController selects and clears index 2`] = `
<div>
  <div><span id="s0"></span><span id="s1"></span><span id="s2"></span></div>
</div>
`;
